### PR TITLE
Updates to troubleshooting guide from upstream docs (#885)

### LINKED ISSUE #885: Inclued additions from upstream PRs
Inspect and include useful info from the following upstream PRs:

- [x] https://github.com/ceph/ceph/pull/39474
- [x] https://github.com/ceph/ceph/pull/39473
- [x] https://github.com/ceph/ceph/pull/39467
- [x] https://github.com/ceph/ceph/pull/39464

### PR DESCRIPTION
**Pull request by tbazant**
closes #885 

rendered PDF is attached 
[book-storage-troubleshooting_color_en.pdf](https://github.com/SUSE/doc-ses/files/6034703/book-storage-troubleshooting_color_en.pdf)

See the new sections:
2.7 Per-service and per-daemon events
3.13 Disabling automatic deployment of daemons
6.4 Restoring the MONs quorum
6.8 Manually deploying a MGR daemon

### CODE REVIEW

File: xml/bp_troubleshooting_monitors.xml [hunk 1]
```diff
@@ -186,6 +186,76 @@
    <literal>mon.a</literal> has rank 0.
   </para>
  </sect1>
+ <sect1 xml:id="mons-restoring-quorum">
+  <title>Restoring the MONs quorum</title>
+
+  <para>
+   In case the &mon;s cannot form a quorum, &cephadm; is not able to manage the
```
Comment (lproven) [inline xml/bp_troubleshooting_monitors.xml:193]:
```suggestion
   If the &mon;s cannot form a quorum, &cephadm; will not be able to manage the
```

File: xml/bp_troubleshooting_monitors.xml [hunk 2]
```diff
@@ -186,6 +186,76 @@
...
+   cluster until the quorum is restored. In order to restore the MON quorum,
```
Comment (lproven) [inline xml/bp_troubleshooting_monitors.xml:194]:
```suggestion
   cluster until the quorum is restored. In order to restore the &mon; quorum,
```

Comment (lproven) [reply xml/bp_troubleshooting_monitors.xml:194]: Entity here is only a suggestion

File: xml/bp_troubleshooting_monitors.xml [hunk 3]
```diff
@@ -186,6 +186,76 @@
...
+   remove unhealthy MONs form the monmap by following these steps:
```
Comment (lproven) [inline xml/bp_troubleshooting_monitors.xml:195]: Ditto

File: xml/bp_troubleshooting_monitors.xml [hunk 4]
```diff
@@ -186,6 +186,76 @@
...
+  </para>
+
+  <procedure>
+   <step>
+    <para>
+     Stop all MONs. Log in to each MON host via SSH and run the following
```
Comment (lproven) [inline xml/bp_troubleshooting_monitors.xml:201]: Ditto

File: xml/bp_troubleshooting_monitors.xml [hunk 5]
```diff
@@ -186,6 +186,76 @@
...
+     <replaceable>MON_ID</replaceable> is usually identical to the string that
+     the <command>hostname</command> command returns:
+    </para>
+<screen>ceph-mon -i <replaceable>MON_ID</replaceable> --extract-monmap <replaceable>/tmp/monmap</replaceable></screen>
+   </step>
+   <step>
+    <para>
+     Remove the non-surviving or problematic monitors. For example, if you have
+     three monitors, <literal>mon.a</literal>, <literal>mon.b</literal>, and
+     <literal>mon.c</literal> where only <literal>mon.a</literal> will survive,
```
Comment (lproven) [inline xml/bp_troubleshooting_monitors.xml:226]: **will** survive or already **is** surviving? I am not sure of the intention here.

File: xml/bp_troubleshooting_monitors.xml [hunk 6]
```diff
@@ -186,6 +186,76 @@
...
+    <para>
+     Start only the surviving monitors, and verify that the monitors form a
+     quorum with the <command>ceph -s</command> command.
+    </para>
+   </step>
+  </procedure>
+
+  <note>
+   <para>
+    You may wish to archive the removed monitors' data directory in
```
Comment (lproven) [inline xml/bp_troubleshooting_monitors.xml:252]:
```suggestion
    You may wish to archive the removed monitors' data directories from
```

File: xml/bp_troubleshooting_monitors.xml [hunk 7]
```diff
@@ -204,15 +274,15 @@ mon.a (rank 0) addr 127.0.0.1:6789/0 is down (out of quorum)</screen>
...
-    problems, check the problematic monitor’s <literal>mon_status</literal>
-    via the admin socket. Considering the monitor is out of the quorum, its
-    state should be one of <literal>probing</literal>,
-    <literal>electing</literal> or <literal>synchronizing</literal>. If it
-    happens to be either <literal>leader</literal> or <literal>peon</literal>,
-    then the monitor believes to be in quorum, while the remaining cluster is
-    sure it is not; or maybe it got into the quorum while we were
-    troubleshooting the monitor, check <command>ceph -s</command> again just to
-    make sure. Continue if the monitor is not yet in quorum.
+    problems, check the problematic monitor’s <literal>mon_status</literal> via
```
Comment (lproven) [inline xml/bp_troubleshooting_monitors.xml:277]:
```suggestion
    problems, check the problematic monitor's <literal>mon_status</literal> via
```

File: xml/bp_troubleshooting_monitors.xml [hunk 8]
```diff
@@ -204,15 +274,15 @@ mon.a (rank 0) addr 127.0.0.1:6789/0 is down (out of quorum)</screen>
...
+    the admin socket. Considering the monitor is out of the quorum, its state
+    should be one of <literal>probing</literal>, <literal>electing</literal> or
+    <literal>synchronizing</literal>. If it happens to be either
+    <literal>leader</literal> or <literal>peon</literal>, then the monitor
+    believes to be in quorum, while the remaining cluster is sure it is not; or
```
Comment (lproven) [inline xml/bp_troubleshooting_monitors.xml:282]:
```suggestion
    believes itself to be in the quorum, while the remaining cluster is sure it is not; or
```

File: xml/bp_troubleshooting_monitors.xml [hunk 9]
```diff
@@ -204,15 +274,15 @@ mon.a (rank 0) addr 127.0.0.1:6789/0 is down (out of quorum)</screen>
...
+    maybe it got into the quorum while we were troubleshooting the monitor,
+    check <command>ceph -s</command> again just to make sure. Continue if the
```
Comment (lproven) [inline xml/bp_troubleshooting_monitors.xml:284]:
```suggestion
    maybe it got into the quorum while we were troubleshooting the monitor.
    Check using <command>ceph -s</command> again, just to make sure. Continue if the
```

File: xml/bp_troubleshooting_monitors.xml [hunk 10]
```diff
@@ -630,8 +700,8 @@ debug ms = 1
     </listitem>
     <listitem>
      <para>
-      If you have no quorum, use the monitor’s admin socket and directly
-      adjust the configuration options:
+      If you have no quorum, use the monitor’s admin socket and directly adjust
```
Comment (lproven) [inline xml/bp_troubleshooting_monitors.xml:703]:
```suggestion
      If you have no quorum, use the monitor's admin socket and directly adjust
```

**Review by lproven — CHANGES_REQUESTED** [review]

**Review by jecluis — APPROVED** [review]

File: xml/bp_troubleshooting_cephadm.xml [hunk 11]
```diff
@@ -484,4 +484,59 @@ Failed to connect to node5 (node5.example.com).
...
+
+  <para>
+   By default, &ceph; daemons are automatically deployed to new hosts after you
+   add these host to the placement specification and apply it (refer to
+   <xref linkend="deploy-cephadm-day2"/> for more details).
+  </para>
+
+  <para>
+   If you need to disable the automated deployment of &ceph; daemons, add
+   <literal>unmanaged: true</literal> to its specification and apply it, for
```
Comment (tbazant) [inline xml/bp_troubleshooting_cephadm.xml:498]:
```suggestion
   <literal>unmanaged: true</literal> to its specification file and apply it, for
```

File: xml/bp_troubleshooting_logging.xml [hunk 12]
```diff
@@ -1487,4 +1487,46 @@ debug mds balancer = 1/20
 
 <screen>&prompt.root;echo "file fs/ceph/snap.c -p" > /sys/kernel/debug/dynamic_debug/control</screen>
  </sect1>
+ <sect1 xml:id="logging-service-daemons-events">
+  <title>Per-service and per-daemon events</title>
+
+  <para>
+   In order to simplify debugging deployments of failed daemons, &cephadm;
```
Comment (tbazant) [inline xml/bp_troubleshooting_logging.xml:1494]:
```suggestion
   In order to simplify debugging failed daemon deployments, &cephadm;
```

File: xml/bp_troubleshooting_monitors.xml (hunk 3 above)
Comment (tbazant) [inline xml/bp_troubleshooting_monitors.xml:195]:
```suggestion
   remove unhealthy &mon;s form the monmap by following these steps:
```

File: xml/bp_troubleshooting_monitors.xml (hunk 4 above)
Comment (tbazant) [inline xml/bp_troubleshooting_monitors.xml:201]:
```suggestion
     Stop all &mon;s. Log in to each &mon; host via SSH and run the following
```

File: xml/bp_troubleshooting_monitors.xml (hunk 5 above)
Comment (tbazant) [inline xml/bp_troubleshooting_monitors.xml:226]:
```suggestion
     <literal>mon.c</literal> where only <literal>mon.a</literal> is surviving,
```

**Pull request merged**